selfupdate/update_test.go

```diff
@@ -11,16 +11,34 @@ import (
 	"github.com/blang/semver"
 )
 
-func setupTestBinary() {
-	if err := exec.Command("go", "build", "./testdata/github-release-test/").Run(); err != nil {
+func setupTestBinary(name ...string) {
```

Comment:
The changes in here and `teardownTestBinary` feel a bit hacky. An alternative I considered was using a new repo (i.e `rhysd-test/test-release-different-name`) which would have contained an archive with a differently named binary, and thus we could leave these two methods here as-is.

Do you have a preference @rhysd ?